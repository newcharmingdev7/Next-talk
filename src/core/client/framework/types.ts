// TODO: (@cvle) Extract useful common types into its own package.
export { Omit, Overwrite, PropTypesOf } from "coral-ui/types";
export { DeepPartial } from "coral-common/types";

export type RelayEnumLiteral<T> = keyof T | "%future added value";
