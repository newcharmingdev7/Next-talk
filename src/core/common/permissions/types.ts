export type UserRole =
  | "%future added value"
  | "COMMENTER"
  | "MEMBER"
  | "STAFF"
  | "MODERATOR"
  | "ADMIN";

export interface User {
  id: string;
  role: UserRole;
  moderationScopes?: null | {
    scoped?: boolean;
    siteIDs?: string[];
  };
}

export type Moderator = User & {
  role: "MODERATOR";
};

export type LTEModerator = User & {
  role: "MODERATOR" | "STAFF" | "MEMBER" | "COMMENTER";
};

export const isModerator = (user: User): user is Moderator => {
  return user.role === "MODERATOR";
};

export const isLTEModerator = (user: User): user is LTEModerator => {
  return user.role !== "ADMIN";
};
