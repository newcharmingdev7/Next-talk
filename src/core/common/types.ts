export type Diff<T extends keyof any, U extends keyof any> = ({ [P in T]: P } &
  { [P in U]: never } & { [x: string]: never })[T];

export type Omit<U, K extends keyof U> = Pick<U, Diff<keyof U, K>>;

export type Overwrite<T, U> = Pick<T, Diff<keyof T, keyof U>> & U;

export type Sub<T, U> = Pick<T, Diff<keyof T, keyof U>>;

/**
 * Make all properties in T writeable
 */
export type Writeable<T> = { -readonly [P in keyof T]: T[P] };

export type Promiseable<T> = Promise<T> | T;
