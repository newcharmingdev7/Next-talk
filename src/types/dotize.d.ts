declare module "dotize" {
  export function convert(obj: Object): Record<string, any>
}
