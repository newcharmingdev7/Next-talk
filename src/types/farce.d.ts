declare module "farce" {
  export class BrowserProtocol {}
  export class queryMiddleware {}
}
