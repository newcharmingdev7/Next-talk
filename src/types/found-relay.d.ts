declare module "found-relay" {
  import { Environment } from "relay-runtime";
  export class Resolver {
    constructor(relayEnvironment: Environment);
  }
}
