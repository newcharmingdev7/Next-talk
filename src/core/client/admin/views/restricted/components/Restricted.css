.lockIcon {
  color: var(--palette-grey-dark);
}
.noPermission {
  color: var(--palette-grey-dark);
}
.contactAdmin {
  color: var(--palette-grey-main);
}
