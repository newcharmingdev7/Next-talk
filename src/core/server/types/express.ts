import { Request } from "express";

import { Tenant } from "talk-server/models/tenant";
import { User } from "talk-server/models/user";
import TenantCache from "talk-server/services/tenant/cache";

export interface Request extends Request {
  user?: User;
  tenant?: Tenant;
  tenantCache: TenantCache;
}
