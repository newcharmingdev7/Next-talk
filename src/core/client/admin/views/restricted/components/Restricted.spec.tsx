import { noop } from "lodash";
import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import { PropTypesOf } from "talk-framework/types";

import Restricted from "./Restricted";

it("renders correctly", () => {
  const props: PropTypesOf<typeof Restricted> = {
    username: "User",
    onSignInAs: noop,
  };
  const renderer = createRenderer();

  renderer.render(<Restricted {...props} />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
