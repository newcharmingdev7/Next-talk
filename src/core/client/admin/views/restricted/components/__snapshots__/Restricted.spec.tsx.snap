// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<AuthBox
  title={
    <Localized
      id="restricted-currentlySignedInTo"
    >
      <span>
        Currently signed in to
      </span>
    </Localized>
  }
>
  <withPropsOnChange(HorizontalGutter)
    size="double"
  >
    <div>
      <withPropsOnChange(Flex)
        justifyContent="center"
      >
        <withPropsOnChange(Icon)
          className="Restricted-lockIcon"
          size="lg"
        >
          lock
        </withPropsOnChange(Icon)>
      </withPropsOnChange(Flex)>
      <Localized
        id="restricted-noPermissionInfo"
      >
        <withPropsOnChange(Typography)
          align="center"
          className="Restricted-noPermission"
          variant="heading3"
        >
          You do not have permission to access this page.
        </withPropsOnChange(Typography)>
      </Localized>
    </div>
    <div>
      <Localized
        id="restricted-signedInAs"
        username={<Username />}
      >
        <withPropsOnChange(Typography)
          align="center"
          container="div"
          variant="bodyCopy"
        >
          You are signed in as: &lt;username&gt;&lt;/username&gt;
        </withPropsOnChange(Typography)>
      </Localized>
    </div>
    <withPropsOnChange(Flex)
      justifyContent="center"
    >
      <Localized
        id="restricted-signInWithADifferentAccount"
      >
        <withPropsOnChange(Button)
          color="primary"
          onClick={[Function]}
          variant="filled"
        >
          Sign in with a different account
        </withPropsOnChange(Button)>
      </Localized>
    </withPropsOnChange(Flex)>
    <Localized
      id="restricted-contactAdmin"
    >
      <withPropsOnChange(Typography)
        align="center"
        className="Restricted-contactAdmin"
        variant="bodyCopy"
      >
        If you think this is an error, please contact your administrator for assistance.
      </withPropsOnChange(Typography)>
    </Localized>
  </withPropsOnChange(HorizontalGutter)>
</AuthBox>
`;
